lib/matplotlib/axis.py

```diff
@@ -1434,6 +1434,13 @@ def update_units(self, data):
         if converter is None:
             return False
 
+        if (self.converter is not None) and (self.converter != converter):
+            raise TypeError('Attempting to plot data that is '
+                            'registered to be converted with %s but '
+                            'is incompatible with existing axis data '
+                            'converter %s' % (converter.__class__,
+                                self.converter.__class__))
+
         neednew = self.converter != converter
```

Comment:
This line can change to `neednew = self.converter == None`, and maybe change the variable name `neednew` to `needupdate` too since that's really what it means now.